metricbeat/tests/system/test_docker.py

```diff
@@ -0,0 +1,177 @@
+        proc.check_kill_and_wait()
+
+        # Ensure no errors or warnings exist in the log.
+        log = self.get_log()
+        self.assertNotRegexpMatches(log.replace("WARN EXPERIMENTAL", ""), "ERR|WARN")
+
+        output = self.read_output_json()
+        evt = output[0]
+
+        evt = self.clean_fields(evt)
+
+        self.assert_fields_are_documented(evt)
+
+    @unittest.skipUnless(metricbeat.INTEGRATION_TESTS, "integration test")
+    def test_info_fields(self):
+        """
+        test info fields
+        """
+        self.render_config_template(modules=[{
+            "name": "docker",
+            "metricsets": ["info"],
+            "hosts": ["localhost"],
+            "period": "1s"
+        }])
+
+        proc = self.start_beat()
+        self.wait_until(lambda: self.output_lines() > 0, max_timeout=30)
+        proc.check_kill_and_wait()
+
+        # Ensure no errors or warnings exist in the log.
+        log = self.get_log()
+        self.assertNotRegexpMatches(log.replace("WARN EXPERIMENTAL", ""), "ERR|WARN")
+
+        output = self.read_output_json()
+        evt = output[0]
+
+        self.assert_fields_are_documented(evt)
+
+    @unittest.skipUnless(metricbeat.INTEGRATION_TESTS, "integration test")
+    def test_memory_fields(self):
+        """
+        test memory fields
+        """
+        self.render_config_template(modules=[{
+            "name": "docker",
+            "metricsets": ["memory"],
+            "hosts": ["localhost"],
+            "period": "1s"
+        }])
+
+        proc = self.start_beat()
+        self.wait_until(lambda: self.output_lines() > 0, max_timeout=30)
+        proc.check_kill_and_wait()
+
+        # Ensure no errors or warnings exist in the log.
+        log = self.get_log()
+        self.assertNotRegexpMatches(log.replace("WARN EXPERIMENTAL", ""), "ERR|WARN")
+
+        output = self.read_output_json()
+        evt = output[0]
+
+        evt = self.clean_fields(evt)
+        self.assert_fields_are_documented(evt)
+
+    @unittest.skipUnless(metricbeat.INTEGRATION_TESTS, "integration test")
+    def test_network_fields(self):
+        """
+        test info fields
+        """
+        self.render_config_template(modules=[{
+            "name": "docker",
+            "metricsets": ["network"],
+            "hosts": ["localhost"],
+            "period": "1s"
+        }])
+
+        proc = self.start_beat()
+        self.wait_until(lambda: self.output_lines() > 0, max_timeout=30)
+        proc.check_kill_and_wait()
+
+        # Ensure no errors or warnings exist in the log.
+        log = self.get_log()
+        self.assertNotRegexpMatches(log.replace("WARN EXPERIMENTAL", ""), "ERR|WARN")
+
+        output = self.read_output_json()
+        evt = output[0]
+
+        evt = self.clean_fields(evt)
+        self.assert_fields_are_documented(evt)
+
+    def clean_fields(self, evt):
+        # TODO: How to handle lables and ports?
```

Comment:
I prefer to skip this for the moment as I wanted to look into how we can perhaps change the ports structure. But I agree, having something like described above could not only be useful here.